s3stream/src/main/java/com/automq/stream/s3/wal/impl/object/ObjectWALService.java

```diff
@@ -36,17 +39,16 @@
 import java.util.Objects;
 import java.util.Queue;
 import java.util.concurrent.CompletableFuture;
-
 import org.apache.commons.lang3.exception.ExceptionUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import static com.automq.stream.s3.wal.common.RecordHeader.RECORD_HEADER_SIZE;
-import static com.automq.stream.s3.wal.common.RecordHeader.RECORD_HEADER_WITHOUT_CRC_SIZE;
 
 public class ObjectWALService implements WriteAheadLog {
     private static final Logger log = LoggerFactory.getLogger(ObjectWALService.class);
 
+    protected FixedSizeByteBufPool byteBufPool = new FixedSizeByteBufPool(RECORD_HEADER_SIZE, 1024);
```

Comment:
When the server is handling a large amount of requests, 1024 headers may not be sufficient. Consider increasing the pool size as the number of CPU cores increases. See #1882 for reference.